### ISSUE TITLE
DropdownV2 Accessible name does not contain the visible text label

### ISSUE THREAD

**Issue by ajlohr** [carbon-design-system/carbon#2517]
DropdownV2 aria-label for the div with role=button needs to include the displayed text. Can you add the selected text to the aria-label? Example below.

## Environment

> Operating system

macOS 10.13.3 (17D47)

> Browser

Chrome Version 73.0.3683.75 (Official Build) (64-bit)

> Automated testing tool and ruleset

IBM Accessibility February 2019

> Assistive technology used to verify

DAP

## Detailed description

Dropdown V2 aria-label does not include visible text of selected item. Would it be possible to include the selected items name/value in the aria-label of the div with role=button?

Doing the below cleared the DAP failure for me, so it would be nice for the component to include the selected text in the aria-label

Example-  the DropdownV2 renders the below HTML. DAP wants the `aria-label` in the div with `role="button"` to have an aria-label that includes `defaultLocation` text, because its the displayed text. Currently the aria-label toggles between `open menu` and `close menu`.
```
<div role="button" class="bx--list-box__field" tabindex="0" type="button" aria-label="open menu" aria-expanded="false" aria-haspopup="true" data-toggle="true">

  <span class="bx--list-box__label" for="downshift-0-input">defaultLocation</span>

  <div class="bx--list-box__menu-icon">
    <svg fill-rule="evenodd" height="5" role="img" viewBox="0 0 10 5" width="10" alt="Open menu" focusable="false" aria-label="Open menu">
      <title>Open menu</title>
      <path d="M0 0l5 4.998L10 0z"></path>
    </svg>
 </div>
</div>
```

> What version of the Carbon Design System are you using?

```
        "carbon-addons-data-viz-react": "1.21.0",
        "carbon-components": "9.84.14",
        "carbon-components-react": "6.108.0",
```

> What did you expect to happen?

aria-label to include displayed text

> What happened instead?

aria-label just toggles between open/close menu. 

> What WCAG 2.1 checkpoint does the issue violate?

Accessible name does not contain the visible text label
IBM 2.5.3 Label in Name

## Steps to reproduce the issue

1. Load DropDownV2 and run DAP against it
